### PR TITLE
Read jpeg exif with RandomAccessInputStream

### PR DESCRIPTION
**Pull request by manics**
Requires a bump to metadata-extractor (I've gone for the latest release) since the `2.6.2` only supports `File`.

### CODE REVIEW

**Comment by manics** [comment]
@sbesson You've excluded this, did it break something?

**Comment by sbesson** [comment]
Sorry @manics just realized my review was still in a pending state - see https://github.com/openmicroscopy/bioformats/pull/3252#discussion_r228024275.

**Comment by manics** [comment]
Last commit removes `test-no-xerces`, only `loci.formats.utests.MissingEXIFTest` is affected.

**Comment by melissalinkert** [comment]
Discussion from last time ```metadata-extractor``` upgrade was attempted might still be relevant: https://github.com/openmicroscopy/bioformats/pull/2473

**Comment by manics** [comment]
I've added xerces back as an explicit dependency.

**Comment by sbesson** [comment]
Discussed today during the @openmicroscopy/formats meeting. As we saw with other external dependencies , the primary concern is usually not at the Bio-Formats level but rather the impact on building and deploying rest of our application notably OMERO (see https://github.com/openmicroscopy/bioformats/pull/2938).

Given the work on the OMERO 5.5.0 build system is still ongoing and this dependency upgrade was not flagged as a requirement for Bio-Formats 6.0.0, I propose to postpone its inclusion and schedule it in a follow-up Bio-Formats 6.x.z release. Once Bio-Formats 6.0.0 is out, we can review all excluded PR and start re-including them

File: components/formats-bsd/src/loci/formats/services/EXIFServiceImpl.java [hunk 1]
```diff
@@ -59,20 +59,19 @@ public EXIFServiceImpl() {
...
     checkClassDependency(com.adobe.xmp.XMPMeta.class);
-    // check for xercesImpl.jar
-    checkClassDependency(org.apache.xerces.dom.NodeImpl.class);
   }
 
   // -- EXIFService API methods --
 
   @Override
   public void initialize(String file) throws ServiceException, IOException {
+    RandomAccessInputStream jpegFile = new RandomAccessInputStream(file);
```
Comment (sbesson) [inline components/formats-bsd/src/loci/formats/services/EXIFServiceImpl.java]: This could be simplified into a [try-with-resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) statement, removing the final `.close()` call.

**Comment by sbesson** [comment]
With the work on HCS metadata and file leak being advanced and the repository tests being green, reincluding in the daily builds.

File: components/formats-bsd/src/loci/formats/services/EXIFServiceImpl.java (hunk 1 above)
Comment (sbesson) [reply components/formats-bsd/src/loci/formats/services/EXIFServiceImpl.java]: Actually with #3331, https://web-proxy.openmicroscopy.org/east-ci/job/BIOFORMATS-test-folder/61880/console flagged this leaves open file handles and failed the repository tests as expected. Excluding until this is resolved and relaunching the daily jobs.

**Comment by manics** [comment]
Rebased and updated

**Comment by sbesson** [comment]
A few failures reported for the [li-flim](https://web-proxy.openmicroscopy.org/east-ci/job/BIOFORMATS-test-folder/62245/), [zeiss-zvi](https://web-proxy.openmicroscopy.org/east-ci/job/BIOFORMATS-test-folder/62250/), [nikon](https://web-proxy.openmicroscopy.org/east-ci/job/BIOFORMATS-test-folder/62287/) and [canon](https://web-proxy.openmicroscopy.org/east-ci/job/BIOFORMATS-test-folder/62323/) formats.

For all of these, the failures happen with the accompanying JPEG screenshots (also configured for the testing) with all date being off-by-one-hour. I suspect this is related to upstream changes related to the date parsing - see https://github.com/drewnoakes/metadata-extractor/releases/tag/2.9.0.

**Comment by manics** [comment]
Checked `Learning to Fly.JPG` which has [`Date = 2011-07-11T09:30:18` in `.bioformats`](https://github.com/openmicroscopy/data_repo_config/blob/b246bd7e29ccd1e4996ba89d8db781e4115f6ec0/curated/canon/pete/.bioformats#L186)

https://www.exif.org/Exif2-2.PDF:
> DateTime
>  The  date  and  time  of  image  creation.  In  this  standard  it  is  the  date  and  time  the  file  was  changed.  The  format  is  "YYYY:MM:DD  HH:MM:SS"  with  time  shown  in  24-hour  format,  and  the  date  and  time  separated  by  one  blank  character [20.H]. When the date and time are unknown, all the character spaces except colons (":") may be filled with  blank  characters,  or  else  the  Interoperability  field  may  be  filled  with  blank  characters.  The  character  string  length is 20 bytes including NULL for termination. When the field is left blank, it is treated as unknown.

There is no time-zone information.

`exiftool` suggests the correct datetime is `2011:07:11 10:30:18`
```
# exiftool -time:all /data/Learning\ to\ Fly.JPG
File Modification Date/Time     : 2019:04:08 09:59:24+00:00
File Access Date/Time           : 2019:04:08 10:43:58+00:00
File Inode Change Date/Time     : 2019:04:08 10:16:38+00:00
Modify Date                     : 2011:07:11 10:30:18
Date/Time Original              : 2011:07:11 10:30:18
Create Date                     : 2011:07:11 10:30:18
```

Testing with BF v6.0.1 https://github.com/openmicroscopy/bioformats/blob/v6.0.1/Dockerfile shows that the success of failure of the test depends on the configure timezone.
`docker run -it --rm -v $TESTDIR:/opt/data -v $TESTDIR:/opt/config -e TZ=$TZ bioformats`
- TZ=Europe/London (default): Passes
- TZ=UTC: `DeltaT: FAILED (series 0 (expected 2011-07-11T09:30:18, actual 2011-07-11T10:30:18))`
- TZ=Australia/Tasmania: `DeltaT: FAILED (series 0 (expected 2011-07-11T09:30:18, actual 2011-07-11T00:30:18))`

**Comment by manics** [comment]
Timestamps have been updated in https://github.com/openmicroscopy/data_repo_config/pull/391
